### ISSUE TITLE
Documents in Minimongo corrupted when using View in subscription

### ISSUE THREAD

**Issue by FlyingTigra** [meteor/meteor#10382]
Hi guys,
- in Meteor 1.8.0_1 
- Mongo 3.4.10
we have a Mongo View that we query in subscription like that:

>const handle = TestView.find({_id: {"$in": [...]}}, options).observeChanges({
>              
> added: function (id, fields) {
>    console.log("Added "+id+", fields "+JSON.stringify(fields));
>        self.added("V_TestView", id, fields);
>    },
>
>    removed: function (id) {
>        console.log("Removed "+id);
>        self.removed("V_TestView", id);
>    },
>
>    changed: function (id, fields) {
>        console.log("Changed "+id+", fields "+JSON.stringify(fields));
>        self.changed("V_TestView", id, fields);
>    }
> })

I see 'added' fired with following set of fields and those fields I see in Minimongo:
>    "daysCd":"xxx"
>     "daypartDesc":"xxx"
>     "broadcastTimeDesc":"xxx"
>     "qtrCd":"xxx"
>     "dealCost":"xxx"
>     "originalDesc":"xxx"
>     "stationCL":"xxx"
>     "desc":"xxx"
>     "dmaCd":"xxx"
>     "timeDesc24":"xxx"
>     "spotLen":xxx
>     "campaignId":"xxx"
>     "timeDesc":"xxx"
>     "cpe":"xxx"
>     "clientCd":"xxx"
>     "prodCd":"xxx"
>     "estCd":"xxx"
>     "demoCd":"xxx"
>     "demo": {}

Then, in 3-5 seconds, I see 'changed' fired on same documents with following set of fields:
> "source": {}, 
> "purchaseHistoryId": "xxx", 
>"nielsenLookupKey": "xxx", 
>"grossRate": xxx

and result document in Minimongo looked like that:

> broadcastTimeDesc: "xxx"
> campaignId: "xxx"
> daypartDesc: "xxx"
> daysCd: "xxx"
> dealCost: xxx
> desc: "xxx"
> dmaCd: "xxx"
> grossRate: xx
> nielsenLookupKey: "xxx"
> originalDesc: "xxx"
> purchaseHistoryId: "xxx"
> qtrCd: "xxx"
> source: {}
> spotLen: xx
> stationCL: "xxx"
> timeDesc: "xxx"
> timeDesc24: "xxx"

So, basically, "demo" and "demoCd" are disappeared from the documents in Minimongo even as 'changed' event didn't have them listed as 'undefined'.

If I add 'demoCd' filter on top of ids, something like
>const handle = TestView.find({_id: {"$in": [...]}, demoCd:"xxx"}, options).observeChanges({...});

then after initial 'added', I see 'removed' and all documents got removed from Minimongo.

I could be wrong, but for me looks like a bug. Please advice.

**Comment by hwillson** [comment]
Any chance you could provide a small runnable reproduction that shows this happening? Thanks!
